src/Chapter2.hs

```diff
@@ -841,7 +850,13 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n l
+  | n < 0 = []
+  | null l = []
+  | otherwise =
+    let len = length l
+    in take len (drop (mod n len) (cycle l))
```

Comment:
Nice and efficient implementation of the function. Great job on covering all corner-cases as well! :medal_sports: